cpp/src/spatial/linestring_distance.cu

```diff
@@ -38,6 +38,48 @@
 namespace cuspatial {
 namespace {
 
+template <typename T>
+vec_2d<T> __device__ operator+(vec_2d<T> const& a, vec_2d<T> const& b)
+{
+  return vec_2d<T>{a.x + b.x, a.y + b.y};
+}
+
+template <typename T>
+vec_2d<T> __device__ operator-(vec_2d<T> const& a, vec_2d<T> const& b)
+{
+  return vec_2d<T>{a.x - b.x, a.y - b.y};
+}
+
+template <typename T>
+vec_2d<T> __device__ operator*(vec_2d<T> const& a, vec_2d<T> const& b)
```

Comment:
Is componentwise multiplication an operator we should have?

Reply:
It's not used in this PR. Can be removed and reintroduced later if needed.

Reply:
Not in the context of this PR. Removed.